# Microevolutionary-dynamics-show-tropical-valleys-are-deeper-for-montane-birds-of-the-Atlantic-Forest
Scripts and datasets necessary to replicate the study

Bioinformatics and genetic structure between and within mountain regions 

Sample demultiplexing, de novo assembly, and SNP calling for each species was performed in ipyrad v0.9.18. Parameter files used in Ipyrad for each species are available at https://github.com/GregoryThom/ and at https://doi.org/10.5281/zenodo.5510615. 
We tested for the number of ancestral populations using conStruct in R v3.6. To obtain pairwise genetic distance matrices we used Adegenet v1.347 in R v3.6.

Estimation of historical demographic parameters
To estimate historical demographic parameters, we used Fastsimcoal v 2.6. An example of parameter files (.est and .tpl) is available at https://github.com/GregoryThom/ and at https://doi.org/10.5281/zenodo.5510615. We used easySFS v1.0 (https://github.com/isaacovercast/easySFS; access date: 05/01/2020) to estimate the projection of SNPs that maximize the number of segregating sites in the Joint SFSs. To estimate the number of segregating sites (S), nucleotide diversity (π), Tajima’s D, and pairwise FST, we used PipeMaster v0.0.9 47 in R v3.6. Script available at https://github.com/GregoryThom/ and at https://doi.org/10.5281/zenodo.5510615, “1_summary_stats_pipemaster.R”.

Species distribution models 
We reduced sampling biases and homogenized the density of occurrences across space, by applying a spatial thinning using a 10 km distance in spThin v0.1.0.1. Current and past habitat suitability was estimated in Maxent v3.4.1. To avoid model overfitting, we evaluated distinct feature classes and regularization multipliers using the ENMeval v0.3.0. Variables were normalized to the mean and submitted to a PCA as implemented in prcomp v3.6.1. To remove extrapolation from the projected SDMs generated by combinations of climatic variables not represented by the training dataset, we used a multivariate environmental similarity surface (MESS) analysis in dismo v1.3-3. All programs mentioned above were used in R v3.6. The pipelines using these programs is available at https://github.com/GregoryThom/ and at https://doi.org/10.5281/zenodo.5510615, “1_ENM_in_R_CHELSA.R” and “1_ENM_past in_R_CHELSA.R”. 

Occupied environmental space
Values of the CHELSA bioclimatic variables at 30 seconds resolution were extracted from occurrence localities in raster v2.6-7. Variables were normalized and submitted to a PCA with prcomp v 3.6.3. We computed a per-species multidimensional volume of the occupied climatic space using hypervolume v2.0.12. All programs mentioned above were used in R v3.6. The pipelines using these programs is available at https://github.com/GregoryThom/ and at https://doi.org/10.5281/zenodo.5510615, “1_hypervolume.R”

Testing predictors for the spatial variation in population differentiation 
We built mixed-effect models exploring the relative contribution of temperature seasonality and local elevation using the lme4 in R v3.6.
We assessed the individual and the combined effects of landscape resistance layers on genetic distance using ResistanceGA v4.0 in R v3.6. A script to replicate our ResistanceGA analyses is available at https://github.com/GregoryThom/ and at https://doi.org/10.5281/zenodo.5510615, “1_Resistance_GA.R”
To test for a community-wide predictor of genetic differentiation, we implemented a maximum likelihood population effect model (MLPE) with nlme v3.1 and corMLPE v0.0.2 (https://github.com/nspope/corMLPE) in R v3.6.  A script to replicate these analyses is available at  https://github.com/GregoryThom/ and at https://doi.org/10.5281/zenodo.5510615, “1_LMEM.R”

Testing for associations between the environment and genetic summary statistics.
To test for the association between multiple population genetics summary statistics and environmental variables with mountain regions controlling for the relatedness among species, we applied Phylogenetic Generalized Least Squares (PGLS) models using caper v1.0.1. To test for the phylogenetic signal on the measured variables we calculated Blomberg's K with adiv v2.0. All programs mentioned above were used in R v3.6. A script to replicate these analyses is available at  https://github.com/GregoryThom/ and at https://doi.org/10.5281/zenodo.5510615, “1_PGLS”.

BSSM
To simulate the effect of distance on individuals dispersion we built a bidimensional stepping stone model (BSSM) where each locality represented a population with an assigned effective population size (Ne). Populations were connected through migration (Nm) to two of their nearest neighbors. We simulated summary statistics using msABC  (Pavlidis et al. 2010). We generated goodness-of-fit plots using the gfit function of abc v2.1 in R v3.6. The simulated datasets were used to train a neural network (NN) regression model, using the R v3.6 interface to Keras v2.3 (https://github.com/rstudio/keras). The Bidimensional stepping stone coalescent modeling is a novel software described in our manuscript. The code necessary to replicate our approach is available at https://doi.org/10.5281/zenodo.5512912 (https://github.com/gehara/BiSSM).
