Bioinformatics:
Sample demultiplexing, de novo assembly, and SNP calling for each species was performed in ipyrad v0.9.18. Parameter files used in Ipyrad for each species are available at https://github.com/GregoryThom/Microevolutionary-dynamics-show-tropical-valleys-are-deeper-for-montane-birds-of-the-Atlantic-Forest/tree/main/Ipyrad_data. Outputs from Ipyrad in vcf format will be available in Dryad upon publication. 

Species distribution models:
We reduced sampling biases and homogenized the density of occurrences across space, by applying a spatial thinning using a 10 km distance in spThin v0.1.0.1. Current and past habitat suitability was estimated in Maxent v3.4.1. To avoid model overfitting, we evaluated distinct feature classes and regularization multipliers using the ENMeval v0.3.0. To remove extrapolation from the projected SDMs generated by combinations of climatic variables not represented by the training dataset, we used a multivariate environmental similarity surface (MESS) analysis in dismo v1.3-3. All programs mentioned above were used in R. The pipelines using these programs is available at https://github.com/GregoryThom/Microevolutionary-dynamics-show-tropical-valleys-are-deeper-for-montane-birds-of-the-Atlantic-Forest/tree/main/scripts, “1_ENM_in_R_CHELSA.R” and “1_ENM_past in_R_CHELSA.R”. 

Occupied environmental space:
Values of the CHELSA bioclimatic variables at 30 seconds resolution were extracted from occurrence localities in raster v2.6-7. Variables were normalized and submitted to a PCA with prcomp v 3.6.3. We computed a per-species multidimensional volume of the occupied climatic space using hypervolume v2.0.12. All programs mentioned above were used in R. The pipelines using these programs is available at https://github.com/GregoryThom/Microevolutionary-dynamics-show-tropical-valleys-are-deeper-for-montane-birds-of-the-Atlantic-Forest/tree/main/scripts, “1_hypervolume.R”

Testing predictors for the spatial variation in population differentiation:
We assessed the individual and the combined effects of landscape resistance layers on genetic distance using ResistanceGA v4.0. To obtain pairwise genetic distance matrices among individuals across the entire distribution of the species, we estimated Nei’s genetic distance with Adegenet v1.3. All programs mentioned above were used in R. A script to replicate our ResistanceGA analyses is available at https://github.com/GregoryThom/Microevolutionary-dynamics-show-tropical-valleys-are-deeper-for-montane-birds-of-the-Atlantic-Forest/tree/main/scripts, “1_Resistance_GA.R”

To test for a community-wide predictor of genetic differentiation, we implemented a maximum likelihood population effect model (MLPE) with nlme v3.1 and corMLPE v0.0.2 (https://github.com/nspope/corMLPE).  All programs mentioned above were used in R. A script to replicate these analyses is available at https://github.com/GregoryThom/Microevolutionary-dynamics-show-tropical-valleys-are-deeper-for-montane-birds-of-the-Atlantic-Forest/tree/main/scripts, “1_LMEM.R”

Testing for associations between the environment and genetic summary statistics:
To test for the association between multiple population genetics summary statistics and environmental variables with mountain regions controlling for the relatedness among species, we applied Phylogenetic Generalized Least Squares (PGLS) models using caper v1.0.1. To test for the phylogenetic signal on the measured variables we calculated Blomberg's K 25 with adiv v2.0. All programs mentioned above were used in R. A script to replicate these analyses is available at https://github.com/GregoryThom/Microevolutionary-dynamics-show-tropical-valleys-are-deeper-for-montane-birds-of-the-Atlantic-Forest/tree/main/scripts, “1_PGLS”.

Estimation of historical demographic parameters:
To estimate historical demographic parameters, we used Fastsimcoal v 2.6. An example of parameter files (.est and .tpl) are available at https://github.com/GregoryThom/Microevolutionary-dynamics-show-tropical-valleys-are-deeper-for-montane-birds-of-the-Atlantic-Forest/tree/main/Fastsimcoal_input_model. All the inputs and outputs for this analyses will be available in Dryad upon publication. We used easySFS (https://github.com/isaacovercast/easySFS; access date: 05/01/2020) to estimate the projection of SNPs that maximize the number of segregating sites in the Joint SFSs (considering two populations) for each species. To estimate the number of segregating sites (S), nucleotide diversity (π), Tajima’s D, and pairwise FST, we used PipeMaster v0.0.9 47 in R. Script available at https://github.com/GregoryThom/Microevolutionary-dynamics-show-tropical-valleys-are-deeper-for-montane-birds-of-the-Atlantic-Forest/tree/main/scripts, “1_summary_stats_pipemaster.R”.
